fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java

```diff
@@ -2635,10 +2635,19 @@ private String getRequestBaseUri(String type) throws Exception {
     /**
      * Builds a collection of properties that will be passed to the persistence interceptors.
      *
+     * @param type
+     *            the resource type
+     * @param id
+     *            the resource logical ID
+     * @param version
+     *            the resource version
+     * @param searchContext
+     *            the request search context
+     * @return a map of persistence event properties
      * @throws FHIRPersistenceException
      */
     private Map<String, Object> buildPersistenceEventProperties(String type, String id,
-            String version, Map<String, String> requestProperties) throws FHIRPersistenceException {
```

Comment:
at first i was worried about this, but i see that it was never used within the method and should be safe to remove.

the requestProperties argument to our various FHIRRestHelper methods has some interesting history.  Back in the old days we didn't have this anywhere.
then, for a specific solution, they wanted a way to pass different HTTP header values to different entries within the *same* batch/transaction bundle.
we were sort of forced to accommodate that and we did so by:
1. introducing this requestProperties parameter all over
2. defining an extension that clients could set at the batch/transaction bundle entry level for setting/overriding specific "header" values
3. adding logic to set the requestProperties from these extensions with this to something non-null while we process the batch/transaction

when we went to open source, I think we lost parts 2 and 3 of this since it was deemed specific to that one client.  requestProperties lingers in our FHIRRestHelper methods, but i'm not sure whether it ever gets set at all at this point.  I'll open an issue to look into that...we'll probably need to decide whether to:
A. bring back that bundle-entry HTTP headers feature; or
B. remove requestProperties from all over

None of that needs to happen now, just thought I'd share the history :-)

Reply:
opened https://github.com/IBM/FHIR/issues/1852 for it